src/snowflake/telemetry/_internal/exporter/otlp/proto/log_exporter/__init__.py

```diff
@@ -0,0 +1,170 @@
+        attributes[FILEPATH_ATTRIBUTE] = record.pathname
+        attributes[FUNCTION_NAME_ATTRIBUTE] = record.funcName
+
+        # Temporarily storing logger's name in record's attributes.
+        # This attribute will be removed by the emitter.
+        #
+        # TODO(SNOW-975220): Upgrade to OpenTelemetry 1.20.0 or later
+        # and use OpenTelemetry's LoggerProvider.
+        attributes[LOGGER_NAME_TEMP_ATTRIBUTE] = record.name
+        return attributes
+
+    def _translate(self, record: logging.LogRecord) -> _logs.LogRecord:
+        otel_record = super(SnowflakeLoggingHandler, self)._translate(record)
+        otel_record.severity_text = SnowflakeLoggingHandler._getSnowflakeLogLevelName(record.levelname)
+        return otel_record
+
+
+class _SnowflakeTelemetryLogEmitter(_logs.LogEmitter):
+    """A log emitter which creates an InstrumentationScope for each logger name it encounters."""
+
+    def __init__(
+            self,
+            resource: resources.Resource,
+            multi_log_processor: typing.Union[
+                _logs.SynchronousMultiLogProcessor, _logs.ConcurrentMultiLogProcessor
+            ],
+            instrumentation_scope: otel_instrumentation.InstrumentationScope,
+    ):
+        super(_SnowflakeTelemetryLogEmitter, self).__init__(resource, multi_log_processor, instrumentation_scope)
+        self._lock = threading.Lock()
+        self.cached_scopes = {}
+
+    def emit(self, record: _logs.LogRecord):
+        if LOGGER_NAME_TEMP_ATTRIBUTE not in record.attributes:
+            # The record doesn't contain our custom attribute with a logger name,
+            # so we can call the superclass's `emit` method. It will emit a log
+            # record with the default instrumentation scope.
+            super(_SnowflakeTelemetryLogEmitter, self).emit(record)
+            return
+
+        # Creating an InstrumentationScope for each logger name,
+        # and caching those scopes.
+        logger_name = record.attributes[LOGGER_NAME_TEMP_ATTRIBUTE]
+        del record.attributes[LOGGER_NAME_TEMP_ATTRIBUTE]
+        with self._lock:
+            if logger_name in self.cached_scopes:
+                current_scope = self.cached_scopes[logger_name]
+            else:
+                current_scope = otel_instrumentation.InstrumentationScope(logger_name)
+                self.cached_scopes[logger_name] = current_scope
+                
+        # Emitting a record with a scope that corresponds to the logger
+        # that logged it. NOT calling the superclass here for two reasons:
+        #  1. LogEmitter.emit takes a LogRecord, not LogData.
+        #  2. It would emit a log record with the default instrumentation scope,
+        #     not with the scope we want.
+        log_data = _logs.LogData(record, current_scope)
+        self._multi_log_processor.emit(log_data)
+
+        # Remembering the fact that we emitted a log record.
+        global _log_records_emitted
+        _log_records_emitted = True
```

Comment:
Why do we need this, and how would we access? If you need this information you can have it in your own LogWriter not important to be in the `SnowflakeLoggingHandler`